Fix/release m4 review

stroncoso-quobis opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* correction
* cleanup

#### What this PR does / why we need it:

Latests comments from M3 review and M4 review preparations

#### Which issue(s) this PR fixes:

Fixes #70 

#### Special notes for reviewers:

- No functionality changes further than Commonalities public release
- Specific comments and insights at #70 and #62 

#### Changelog input

```
release-note
- fix(readme): terminology and scope review. Removed virtual meeting url
- fix(*): Removed 5xx generic responses
- fix(*): OAS review: descriptions, terminology, missing elements, paths …
- fix(regsitration): PUT 200 no content replaced by 204 no content
- fix(*): Ensure error 400, 401, 403 responses are included
- fix(*): Replaced "racm", "vvoip" session sessionId suffix
- fix(*): replaced transactionId by x-correlator CAMARA transaction identifier

```

#### Additional documentation 

```
docs
- doc(*): Inlcuded basic error testing (400, 401, 404)
- doc(*): Updated API-Readiness files
```

stroncoso-quobis commented:
Hi team,
Thanks to @pradeepachar-mavenir , for your review,

Inform about a couple of changes pending, that are already in progress:
- API readiness update
  - https://github.com/camaraproject/WebRTC/issues/70#issuecomment-2668594383
  - Just basic identifier changes, already done, pending to push.
- Feature testing error codes for all APIs
  - https://github.com/camaraproject/WebRTC/issues/70#issuecomment-2668634648
  - Without 5xx errors, it is easy to cover 400, 401, 403, 404. But it takes a little bit more time.
  - Already covered webrtc-call-handling, pending to push. Pending webrtc-registration and webrtc-call-handling.

And pending to analyze requirements for webrtc-events (subscription API) due M4 public release of commonalities.

As soon as they are ready, I will mark the PR as ready to merge.
Tomorrow, at working group meeting we will discuss changes to cover here.

Best regards,

stroncoso-quobis commented:
All points identified at #70 should be fixed. Some extra corrections included.

We replaced some terms like "racm" and "vvoip" to keep a clearer syntax.

hdamker commented:
@stroncoso-quobis There are two options:

1. This is also already the release PR for M4 with r1.2 (and with the versions 0.2.0 instead of 0.2.0-rc.1, respectively 0.1.0): for that you need to update the release and version numbers accordingly across the files and add the CHANGELOG.md update
2. You plan to do the actual release PR separately and just want to get a review on the changes: but then you need to reset the version across the files to "wip" as there is already a released version 0.2.0-rc.1

Given the short time until Friday I recommend to go for option 1.

cc: @tanjadegroot

stroncoso-quobis commented:
Hi team,

@hdamker , @tanjadegroot , @pradeepachar-mavenir

Proceeding with requested changes, yes, as suggested by Herbert, we are pointing to M4 release. So summarizing changes from latest comment:

* Commit: d2fe740fbfbcd9ded062a36500bb6dce82a343e4
  * doc(*): Remove rc.X suffix from version, targeting M4 release
  * Updates to OAS `yaml`, API-Readiness `md`, and test_defintions (`feature`)
* Commit: fdcb63451625fd204d43a6ceb3fd6b41e1d057d0
  * Updating webrtc-events to be aligned with commonalities 0.5.0 (rc2.2)
  * Follow comments and review points as https://github.com/camaraproject/ReleaseManagement/issues/175
  * Mainly targeting webrtc-events API (susbcription API):
    * UTC timestamp description on AccessExpireUtc
    * Sink uri update
    * 409 response review
    * SinckCredentials removed from responses
* Commit: 11350398fc86562a86ff751cbf6b553aef712180
  * doc: README and CHANGELOG updates for r1.2
  * Review of CHANGELOG: Cover changes from r1.1 to r1.2 (present PR)

Last commit for trailing spaces :facepalm: 

Hope everything is ok. Waiting for your reviews

Best regards,

tanjadegroot commented on code/API_definitions/webrtc-events.yaml:
```diff
@@ -586,12 +599,12 @@ components:
           description: |-
-            The RACM session ID created by WebRTC Gateway.
-            Obtained via RACM API.
+            The resgitration session ID created by WebRTC Gateway.
```
```suggestion
            The registration session ID created by the network.
```

tanjadegroot commented on code/API_definitions/webrtc-events.yaml:
```diff
@@ -852,9 +871,9 @@ components:
+        mediaSessionId:
           type: string
-          description: The VVOIP session ID created by WebRTC Gateway. The vvoipSessionId shall not be included in POST requests by the client, but must be included in the notifications from the WebRTC GW to client.
+          description: The VVOIP session ID created by WebRTC Gateway. The mediaSessionId shall not be included in POST requests by the client, but must be included in the notifications from the WebRTC GW to client.
```
VVoIP --> media

tanjadegroot commented on code/API_definitions/webrtc-events.yaml:
```diff
@@ -1358,16 +1377,16 @@ components:
 
-        Identified by for `vvoipSessionId`, and validated via a `racmSessionId`
+        Identified by for `mediaSessionId`, and validated via a `regSessionId`
         the endpoint is able to interact with its own voice-video session.
```
voice-video --> media

tanjadegroot commented on code/API_definitions/webrtc-events.yaml:
```diff
@@ -1393,7 +1412,7 @@ components:
         A new WebRTC session is requested (i.e. incoming call). This event
```
WebRTC --> media
comment applicable to the whole file

stroncoso-quobis commented on code/API_definitions/webrtc-events.yaml:
Mostly of stanadlone "webrtc" usage was replaced with "media", there are still some usages still available, since they make sense to be consistent with the API name "webrtc-events". But, in general, the term "media session" will replace all "webrtc session" usages.

Thanks

stroncoso-quobis commented on code/API_definitions/webrtc-registration.yaml:
```diff
@@ -200,32 +240,23 @@ components:
       properties:
         deviceId:
           type: string
           description: The device-id of the client in UUID format.
```
Anayway, we will consider to include a phoneNumber (tel URI) or NetworkIdentifier (SIP URI)... it makes sense.

tanjadegroot reviewed (changes requested):
Many thanks for the additional updates. 

Can you please still clarify on the access token part (see https://github.com/camaraproject/WebRTC/pull/71#discussion_r1976091066)

Also there seem to be some errors (trailing spaces) to be fixed in webrtc-registration.yaml

stroncoso-quobis commented on code/API_definitions/webrtc-registration.yaml:
```diff
@@ -108,67 +106,109 @@ servers:
+      summary: Refresh access token and network registration.
+      description: |
+        API Endpoint used to refresh the resgitration on the network. Sharing a
+        new AccessToken with network if it corresponds. The AccessToken
```
Yeah, that's correct, re-registration or registration refresh is required to keep the device enabled on the network, it should not be required to mix with the the access token. It will trigger a netowrk register on the IMS network, I will try to clarify.

stroncoso-quobis commented on code/Test_definitions/webrtc-call-handling-createSession.feature:
```diff
@@ -1,11 +1,11 @@
     And the header "Authorization" is set to a valid access token
-    And the header "transactionId" is set to a UUID value
+    And the header "x-correlator" is set to a UUID value
     And the header "clientId" is set to a UUID value
```
Discussion ongoing, probably we will end deleting it completly.
`clientId` is unique per `registrationId`+`deviceId`, so yeah, in pratical terms, it does not differ from a `registrationId` that must be unique per registration instance (even if it is done on the same device).

As said, discussion ongoing. We will see.

stroncoso-quobis commented:
OK, let's try to summarize all changes for this M4 review at e367dd4bcc1362b19e61669e77d19240b8ad58fd

- Removed rc.x
- Replaced "vvoip" references for "media"
- Replaced "webrtc gw" references for "network"
- Removed references to NotificationChannel
- Improved terms and definitions
- Improved deviceId descriptions
- Removed access token reference on PUT registration
- CHANGELOG reviewed
- Trailing spaces and spelling errors

Find a readable documentation at:

- webrtc-registration: [redoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/WebRTC/e367dd4bcc1362b19e61669e77d19240b8ad58fd/code/API_definitions/webrtc-registration.yaml&nocors) [swagger](https://editor.swagger.io/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fcamaraproject%2FWebRTC%2Fe367dd4bcc1362b19e61669e77d19240b8ad58fd%2Fcode%2FAPI_definitions%2Fwebrtc-registration.yaml)
- webrtc-call-handling: [redoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/WebRTC/e367dd4bcc1362b19e61669e77d19240b8ad58fd/code/API_definitions/webrtc-call-handling.yaml&nocors) [swagger](https://editor.swagger.io/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fcamaraproject%2FWebRTC%2Fe367dd4bcc1362b19e61669e77d19240b8ad58fd%2Fcode%2FAPI_definitions%2Fwebrtc-call-handling.yaml)
- webrtc-events: [redoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/WebRTC/e367dd4bcc1362b19e61669e77d19240b8ad58fd/code/API_definitions/webrtc-events.yaml&nocors) [swagger](https://editor.swagger.io/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fcamaraproject%2FWebRTC%2Fe367dd4bcc1362b19e61669e77d19240b8ad58fd%2Fcode%2FAPI_definitions%2Fwebrtc-events.yaml)

Changes pending for today's discussion at working group meeting:
- notifications: Respond to file deletion request
- registration: Use Device object into request and include testing
- all: remove clientId decission

Regards,

tanjadegroot reviewed (approved):
Looks good ! 

For your consideration: Please check some english and terminology consistency proposals below and 2 questions (also terminology related) and a clarification on the hdrRegistrationId

I approve these changes.  Many thanks for the hard work !
/LGTM from Release Management

stroncoso-quobis commented on code/API_definitions/webrtc-call-handling.yaml:
```diff
@@ -186,24 +195,32 @@ info:
-        - $ref: '#/components/parameters/hdrTransactionId'
-        - $ref: '#/components/parameters/hdrClientId'
+        - $ref: "#/components/parameters/x-correlator"
+        - $ref: '#/components/parameters/hdrRegistrationId'
```
Yes, we were discussing this point during the technical meeting (find notes here: https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/89686183/2025-03-11+WebRTC+Minutes) .

The idea is to use this validator of the scope. To create a media session you need a previous valid registration, so you must identify that registration here, at this header.

stroncoso-quobis commented on code/API_definitions/webrtc-call-handling.yaml:
```diff
@@ -9,16 +9,16 @@ info:
     To handle audio/video sessions, what we usually known as *calls*, the network
     requires to identify both involved parties: caller, the origin, and the callee,
-    the destination. Once identified the endpoints required to be connected, then
+    the destination. Once identified the involved communication endpoints, then
```
Yes, indeed
